### PR TITLE
Property api

### PR DESCRIPTION
**Pull request by kylealexanderhackney**
## Link to JIRA ticket

Migrate existing Rails API to .Net  On Trello

### *What is the problem we're trying to solve*

Adding An Api that matches the swagger docs mentioned in the trello ticket

### *What changes have we introduced*

Controllers and use cases for getting propertioes and alerts.
2 gateways for requesting data from the property and alert platform apis

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [ ] Code pipeline builds correctly

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,31 +2,43 @@
...
-
-.PHONY: shell
-shell:
-	docker-compose run repairs-api bash
-
-.PHONY: test
-test:
-	docker-compose up test-database & docker-compose build repairs-api-test && docker-compose up repairs-api-test
+serve: build
+	docker-compose up repairs-api
```
Comment (LBHBomotoso) [inline Makefile:15]: Not familiar with Makefiles, whats the difference here?

File: RepairsApi/V1/UseCase/ListPropertiesUseCase.cs [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
+
+        public ListPropertiesUseCase(IPropertyGateway propertyGateway)
+        {
+            _propertyGateway = propertyGateway;
+        }
+
+        public Task<IEnumerable<PropertyModel>> ExecuteAsync(PropertySearchModel searchModel)
         {
-            return await Task.FromResult(new List<PropertyModel>()).ConfigureAwait(false);
+            return _propertyGateway.GetByQueryAsync(searchModel);
```
Comment (LBHBomotoso) [inline RepairsApi/V1/UseCase/ListPropertiesUseCase.cs]: Async the way to go?

**Comment by LBHBomotoso** [comment]
Looks fine

File: RepairsApi/V1/UseCase/ListPropertiesUseCase.cs (hunk 2 above)
Comment (kylealexanderhackney) [reply RepairsApi/V1/UseCase/ListPropertiesUseCase.cs]: if we use async then we will not lock threads in the pool while api requests are being made to the platform apis.

**Review by alex-demetriou — APPROVED** [review]
Great stuff :tada:

File: RepairsApi.Tests/RepairsApi.Tests.csproj [hunk 3]
```diff
@@ -1,9 +1,10 @@
-﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+<Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
     <TargetFramework>netcoreapp3.1</TargetFramework>
 
     <IsPackable>false</IsPackable>
     <IsTestProject>true</IsTestProject>
+    <NoWarn>$(NoWarn);CA2007</NoWarn>
```
[the remark this answers is not in the document]
Comment (kylealexanderhackney) [reply RepairsApi.Tests/RepairsApi.Tests.csproj:7]: No it just does the standard removing the warning from being raised when building. This one removes the need for configureawait as its not necessary

**Pull request merged**